torch/distributions/multinomial.py

```diff
@@ -0,0 +1,87 @@
+import torch
+from torch.distributions.distribution import Distribution
+from torch.distributions import Categorical
+from numbers import Number
+from torch.distributions import constraints
+from torch.distributions.utils import log_sum_exp
+
+
+class Multinomial(Distribution):
+    r"""
+    Creates a Multinomial distribution parameterized by `probs` and `total_count`.
+
+    If `probs` is 1D with length-`K`, each element is the relative probability
+    of sampling the class at that index.
+
+    If `probs` is 2D, it is treated as a batch of probability vectors.
```

Comment:
I think your implementation is much more general than this. Consider changing to something like:
> The innermost dimension of `probs` indexes over categories. All other dimensions index over batches.